### PR TITLE
[#2PlaysAMonth]: Travel app

### LINKED ISSUE #419: [Add a Play]: Travel App
Thank You! 

## What's the Play about?
This is a web app that returns the nearest restaurants, hotels and other attractions based on the user's geographical location.

## What ReactJS concept will be used to create this play?
Hooks in React such as useState, UseRefs
Axios in React to fetch data.

## Are you willing to take it up for implementation?
Yes I am. Already working on it

### LINKED ISSUE #1000: [#2PlaysAMonth]: Travel Advisor App
### What's the Play about?

A travel advisor app  where we can checkout locations on  map by  fetching restaurants  and  hotels 

### What ReactJS concept will be used to create this play?

Hooks (will add more in the process of building it)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by yung-coder**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/reactplay/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description

A travel advisor app where we can checkout locations on map by fetching restaurants and hotels

Fixes #1000 

# How Has This Been Tested?

It has been tested by me in my local System

# Checklist:

- [✔️ ] I have performed a self-review of my own code
- [✔️ ] I have commented my code, particularly in hard-to-understand areas
- [✔️ ] I have made corresponding changes to the documentation
- [✔️ ] My changes generate no new warnings
- [✔️ ] I have added tests that prove my fix is effective or that my feature works
- [✔️ ] New and existing unit tests pass locally with my changes
- [✔️ ] Any dependent changes have been merged and published in downstream modules

# Screenshots or example output
none

### CODE REVIEW

**Comment by atapas** [comment]
@yung-coder The attached issue to the PR is closed already. Could you please create a new one and attach?

**Comment by yung-coder** [comment]
> @yung-coder The attached issue to the PR is closed already. Could you please create a new one and attach?

my bad linked wrong issue updated now @atapas

File: src/plays/travel-advisor/styles.css [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/* enter stlyes here */
+
+.Main-Contanier {
```
Comment (Sachin-chaurasiya) [inline src/plays/travel-advisor/styles.css:3]:
```suggestion
.main-contanier {

```

File: src/plays/travel-advisor/Readme.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Travel Advisor 
+
+A travel advisor app where we can checkout locations on map by fetching restaurants and hotels
+
+## Play Demographic
+
+- Language: js
```
Comment (Sachin-chaurasiya) [inline src/plays/travel-advisor/Readme.md:7]:
```suggestion
- Language: JavaScript

```

File: src/plays/travel-advisor/styles/dropdown.css [hunk 3]
```diff
@@ -0,0 +1,52 @@
+select {
```
Comment (Sachin-chaurasiya) [inline src/plays/travel-advisor/styles/dropdown.css:1]: @yung-coder , do not directly change the style of the element, I would recommend using class.

File: src/plays/travel-advisor/styles/dropdown.css [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  margin: 0 0 0.25em;
+}
+a {
+  font-weight: bold;
+  color: var(--gray);
+  text-decoration: none;
+  padding: 0.25em;
+  border-radius: 0.25em;
+  background: white;
+} */
```
Comment (Sachin-chaurasiya) [inline src/plays/travel-advisor/styles/dropdown.css:52]: Remove this.

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]
Left a few comments, thanks

**Comment by yung-coder** [comment]
@Sachin-chaurasiya fixed

**Review by Sachin-chaurasiya — APPROVED** [review]
Great work @yung-coder  🙌🏻,
Thanks for the contributions ✨

**Comment by yung-coder** [comment]
All good to be merged or any more modifications @koustov ....?

**Comment by Sachin-chaurasiya** [comment]
@yung-coder , Build is failing.
image

**Comment by yung-coder** [comment]
Wait a sec @Sachin-chaurasiya

**Comment by yung-coder** [comment]
@Sachin-chaurasiya  done

**Comment by yung-coder** [comment]
@Sachin-chaurasiya  @koustov  fixed linting errors is it again failing build....?

File: src/plays/travel-advisor/styles/dropdown.css [hunk 5]
```diff
@@ -0,0 +1,40 @@
+.select {
```
Comment (koustov) [inline src/plays/travel-advisor/styles/dropdown.css:1]: Please prefix the class names with your play name as you have done for `travel-advisor-main-Container

**Review by koustov — CHANGES_REQUESTED** [review]

**Review by koustov — CHANGES_REQUESTED** [review]
As you are using HTTP call (`http://api.openweathermap.org/geo/1.0/direct?q=london&limit=5&appid=9f66153982a275eb857428a4be592907`) to the API, this will not work on preview and production build. You need to find an HTTPS alternative 
image

**Comment by koustov** [comment]
> @Sachin-chaurasiya @koustov fixed linting errors is it again failing build....?

Build is working now

**Comment by yung-coder** [comment]
@koustov  @atapas  made the required changes ...

**Comment by koustov** [comment]
> @koustov @atapas made the required changes ...

@yung-coder  The preview build is not showing any data. Am I missing anything?

**Comment by koustov** [comment]
> > @koustov @atapas made the required changes ...
> 
> @yung-coder The preview build is not showing any data. Am I missing anything?

I realized its working but 
1. It takes some time to load 
2. only for Indian locations
3. actually, first one is loading but second time if I enter another location, its not firing any call

Anyway, I will be merging it now. You can enhance it later also.

**Pull request merged**